codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/EndpointGenerator.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.go.codegen;
+
+import static software.amazon.smithy.go.codegen.GoWriter.joinWritables;
+
+import java.util.Arrays;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.go.codegen.endpoints.EndpointParametersGenerator;
+import software.amazon.smithy.rulesengine.language.EndpointRuleSet;
+import software.amazon.smithy.utils.SmithyBuilder;
+
+
+public final class EndpointGenerator {
```

Comment:
yah i agree, ill add documentation. but yes, `EndpointGenerator` is a component that will call out to all the classes of the endpoint things for creation. for example, it will eventually look something like this
```go
        parametersGenerator = EndpointParametersGenerator.builder()
                .parametersType(internalOptionsType)
                .build();

        providerGenerator = EndpointProviderGenerator.builder()
                .parametersType(internalOptionsType)
                .resolverType(internalResolverType)
                .newResolverFn(internalNewResolverFn)
                .endpointType(endpointType)
                .resolveEndpointMethodName(resolveEndpointMethodName)
                .build();

        testsGenerator = EndpointTestsGenerator.builder()
                .parametersType(internalOptionsType)
                .resolverType(internalResolverType)
                .newResolverFn(internalNewResolverFn)
                .endpointType(endpointType)
                .resolveEndpointMethodName(resolveEndpointMethodName)
                .build();
```

Reply:
ended up removing this class and letting its functionality be subsumed by `SmithyEndpointGenerator`, and added documentation to that class